calico_containers/calico_ctl/node.py

```diff
@@ -383,7 +431,11 @@ def _start_libnetwork_container(etcd_authority, libnetwork_image):
 
     environment = [
         "HOSTNAME=%s" % hostname,
-        "ETCD_AUTHORITY=%s" % etcd_authority   # etcd host:port
+        "ETCD_AUTHORITY=%s" % etcd_authority,   # etcd host:port
+        "ETCD_SCHEME=%s" % etcd_scheme,         # blank, "http", or "https"
+        "ETCD_KEY_FILE=%s" % etcd_key,          # file path
+        "ETCD_CERT_FILE=%s" % etcd_cert,        # file path
+        "ETCD_CA_CERT_FILE=%s" % etcd_ca_cert,  # file path
```

Comment:
Just a thought - and don't feel you need to agree...

I was wondering if while you compile the etcd_volumes, you also compile a list of etcd_envs and etcd_felix_envs that you could pass in and append to the list.  It would reduce the lines of code and means you wouldn't have to pass in all the individual etcd parms into the _start_*_container methods.  Basically do what you did for the volumes.